src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java

```diff
@@ -182,7 +186,33 @@ private Uni<Supplier<CommandResult>> executeCollectionCreation(QueryExecutor que
               } else {
                 return new SchemaChangeResult(true);
               }
-            });
+            })
+        .onFailure(
+            error ->
+                error instanceof InvalidQueryException
+                    && error
+                        .getMessage()
+                        .matches(
+                            ".*Cannot have more than \\d+ indexes, failed to create index on table.*"))
+        .recoverWithUni(
+            error ->
+                deleteCollectionOperation
+                    .execute(queryExecutor)
+                    .onItem()
+                    .transform(
+                        res ->
+                            ErrorCode.TOO_MANY_INDEXES.toApiException(
+                                "cannot create a new collection; need %d indexes to create the collection;",
+                                dbLimitsConfig.indexesNeededPerCollection()))
+                    .onFailure()
+                    .recoverWithItem(
+                        e -> {
+                          // This is unlikely to happen for delete collection, return with
+                          // TOO_MANY_INDEXES exception
+                          return ErrorCode.TOO_MANY_INDEXES.toApiException(
+                              "collection created with indexes creation failure; need %d indexes to create the collection;",
```

Comment:
Should include collection name somewhere. And maybe reword as "collection creation failed due to index creation failing; need ...  to create a collection"

Reply:
Thanks, solved